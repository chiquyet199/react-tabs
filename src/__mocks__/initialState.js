export const initialState = {
  locale: {
    currentLocale: "en-US",
    translations: {
      common: {
        postcode_lookup: {
          postcode: "Postcode",
          find_address: "Find address",
          address_not_listed: "Address not listed",
        },
      },
      home__welcome: "Welcome to RNW starter app",
      home__loggedin: "You are now logged in",
      login__email__label: "Email address",
      login__password__label: "Password",
      login__email__placeholder: "example@example.com",
      login__password__placeholder: "******",
      login__continue__with__facebook: "Continue with Facebook",
      login: "Sign in",
      create__account: "Create account",
      login__forgot__password: "Forgot password",
      logout: "Logout",
      back: "Back",
      submit: "Submit",
      rent_passport: "Rent Passport",
      deposit_free: "Deposit Free",
      home_management: "Home Management",
      authorize: "Authorize",
      terms_header: "Canopy terms & conditions",
      terms:
        "1. INTRODUCTION\n\nBy signing up for an account with Canopy, you are agreeing to be bound by the following terms and conditions (“Terms of Service”). These Terms of Service are to be read in conjunction with:\n\nour Privacy Policy, which sets out the terms on which we process any personal data we collect from you, or that you or others provide to us; and\n\nour Cookies Policy, which sets out information about the cookies we use.\n\nThese Terms of Service are important. Please read them carefully as they contain details of our statutory and regulatory responsibilities to you, and your contractual obligations to us. They also set out the basis on which we provide, and you are permitted to use, the various services made available on the website www.findyourcanopy.com and the mobile application known as ‘Canopy’ (together, the “Platform”), including the following:",
      verify__email__title__text: "Verify your email address",
      verify__email__text:
        "We need to know you are you before creating a RentPassport. Please click the link in the email we sent you.",
      open__email__client: "Open my email client",
      resend__verification__email: "Resend verification email",
      not__received__an__email: "Not received an email?",
      forgot_password: "Forgot your password?",
      reset_password: "Change your password",
      tenant_reference_submitted: "Reference submitted",
      tenant_reference_thanks_for_time:
        "Thanks for taking the time to submit a tenant reference for",
      tenant_reference_did_you_know:
        "Did you know that Canopy offers the following property management tools to tenants, landlords and agents:",
      tenant_reference_rent_passport_text:
        "Full tenant referencing and property management including deposits and payments",
      tenant_reference_deposit_free_text:
        "Qualified tenants can pay a premium to cover their rental security deposit",
      tenant_reference_home_management_text:
        "With a valid UK Open Banking account, tenants are able to have rent payments contribute to their Experian credit report",
      find_out_more: "Find out more",
      tenant_reference_title: "Tenant reference request",
      upload_tenant_reference_document: "Upload tenant reference document",
      upload_tenant_reference_document_description:
        "If you have a manual form you usually use, please upload it here.",
      upload: "Upload",
      submit_reference: "Submit reference",
      tenant_reference_request: "Tenant reference request",
      complete_reference_form_description:
        " tenant of yours, has authorised us to request a reference from you.\n" +
        "As you know a landlord reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
      upload_reference_document: "Upload reference document",
      complete_reference_form: "Complete reference form",
      complete_reference_form_a: ", a ",
      truelayer: {
        truelayer_auth_title: "Truelayer Authorize",
      },
      employer_reference_form_description:
        " has advised us that you are their employer and has authorised us to request a reference from you.\n" +
        "As you know an employment reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
    },
  },
  auth: {
    user: null,
    isLoggedIn: false,
  },
  dimensions: {
    fontScale: 1,
    height: 901,
    scale: 1,
    width: 1920,
    size: "L",
  },
  inputField: {
    showPassword: true,
  },
  form: {},
  partialForm: {
    values: {},
  },
  progress: 0,
  notification: {
    type: null,
    message: "",
  },
  user: {
    profile: {
      firstName: "FirstName",
      lastName: "LastName",
    },
  },
  router: {
    location: {
      pathname: "/tenant-reference-request/complete",
      search: "",
      hash: "",
    },
    action: "POP",
  },
  lookups: {
    addresses: [],
  },
  topbar: {
    haveTopbar: true,
    topbarBtn: {
      haveTopbarButton: false,
      topbarButtonText: "",
      btnType: "transparent_green",
      topbarButtonIcon: undefined,
      hideBackButton: false,
    },
    topbarCenter: {
      topbarCenterText: "Center",
    },
    topbarLeft: {
      topbarLeftText: "Back",
    },
  },
  images: {
    accepted: [],
  },
  rentPassport: {
    aboutYou: {
      email: "jonathan@gmail.com",
      firstName: "Jonathan",
      lastName: "Beaumont",
      middleName: null,
      gender: "MALE",
      dayOfBirth: 20,
      monthOfBirth: 11,
      yearOfBirth: 1970,
      maritalStatus: "MARRIED",
      phoneNumber: "01123456789",
      nationality: "UK",
      isSmoker: true,
      hasPets: true,
      hasChildren: true,
      status: "GREEN",
      messages: [],
    },
    financial: {
      salary: 100000,
      pension: 0,
      interest: 0,
      benefits: 0,
      rentals: 0,
      other: 0,
      grossAnnualIncome: 100000,
      sourcesOfIncome: ["Employment"],
      status: "YELLOW",
      messages: [],
      bankrupt: false,
      creditScore: 930,
      depositFree: true,
      maximumEligible: 1800.266,
      activeCCJs: 0,
      activeCCJYear: 0,
      activeCCJMonth: 0,
      activeCCJValue: 0,
      previousCCJs: 0,
      previousCCJsYear: 0,
      previousCCJsMonth: 0,
      activeInsolvencies: 0,
      activeInsolvenciesYear: 0,
      activeInsolvenciesMonth: 0,
      previousInsolvencies: false,
      previousInsolvenciesYear: 0,
      previousInsolvenciesMonth: 0,
    },
    rentPassportLegal: {
      pep: false,
      mortality: "passed",
      legalWarnings: false,
      sanctions: false,
      status: "GREEN",
      messages: [],
    },
    whatYouDo: {
      workInfo: [
        {
          workStatus: "EMPLOYED",
          companyName: "companyName",
          jobTitle: "jobTitle",
          income: 100000,
          employmentTime: "FULL_TIME",
          employmentType: "CONTRACT",
          workedSinceMonth: 12,
          workedSinceYear: 2017,
          studentType: "POST_GRADUATE_STUDENT",
          reference: {
            contactFirstName: "Silvester",
            contactLastName: "Stallone",
            contactEmailAddress: "rambo666@gmail.com",
            referredAt: "12-12-2016",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
    whereYouLived: {
      residenceInfo: [
        {
          line1: "Addresss line 1",
          town: "Town",
          state: "State",
          postCode: "UL11223",
          countryCode: "UK",
          residenceType: "RENTING_FROM_AGENT",
          livedSinceMonth: 1,
          livedSinceYear: 2016,
          livedToMonth: null,
          livedToYear: null,
          reference: {
            contactFirstName: "Jonathan",
            contactLastName: "Beaumont",
            contactEmailAddress: "jonathan@gmail.com",
            referredAt: "01-01-2018",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
  },
};
