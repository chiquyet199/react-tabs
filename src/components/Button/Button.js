import styled from "styled-components/native";
import Icon from "../Icon";
import {themed} from "../../utils/common";

export const BtnText = styled.Text`
  position: relative;
  top: -1px;
  font-size: ${({type, theme: {buttons}}) => buttons[type].fontSize || "16px"};
  font-weight: ${({type, theme: {buttons}}) =>
    buttons[type].fontWeight || "normal"};
  color: ${({type, theme: {buttons}}) => buttons[type].text};
`;

export const ButtonPart = styled.View`
  justify-content: center;
  flex: 1;
`;

export const Left = styled(ButtonPart)`
  position: absolute;
  height: 46px;
  width: ${({size}) => `${size}px`};
  left: ${({shift}) => (shift ? "10px" : 0)};
  top: 0;
  bottom: 0;
`;

export const Right = styled(ButtonPart)`
  position: absolute;
  height: 46px;
  width: ${({size}) => `${size}px`};
  right: ${({shift}) => (shift ? "10px" : 0)};
  top: 0;
  bottom: 0;
`;

export const Middle = styled(ButtonPart)`
  align-items: center;
  flex: 1;
  padding: 0;
  padding-right: ${({righticon, lefticon, size}) =>
    righticon || lefticon ? `${size + 10}px` : "10px"};
  padding-left: ${({lefticon, righticon, size}) =>
    lefticon || righticon ? `${size + 10}px` : "10px"};
  min-width: 60px;
`;

const Container = styled.TouchableOpacity`
  border-width: ${props => (props.noBorder ? 0 : 1)};
  width: 100%;
  max-width: 350px;
  margin: 0;
  padding: 0;
  height: 48px;
  border-radius: 6;
  border-color: ${({type, theme: {buttons}}) => buttons[type].border};
  background-color: ${({type, theme: {buttons}}) => buttons[type].background};
  opacity: ${({disabled}) => (disabled ? 0.5 : 1)};
`;

export const ButtonContainer = styled(Container)`
  padding: 0;
  margin: 5px;
  flex: auto;
  max-height: 48px;
  width: 100%;
  align-self: center;
`;

export const StyledIcon = styled(Icon)`
  text-align: center;
`;

export const IconContainer = styled(Container)`
  width: ${({size}) => size || 48}px;
  height: ${({size}) => size || 48}px;
  margin: 0;
  padding: 0;
  align-items: center;
  justify-content: center;
`;

export const MenuButtonIconContainer = styled(IconContainer)`
  margin-top: 0;
  margin-bottom: 2px;
  background-color: transparent;
`;

export const BounceContainer = styled.View`
  position: absolute;
  width: 200px;
  margin-top: ${({theme}) => theme.padding.top}px;
  margin-right: -20px;
  right: 0;
  z-index: 999;
`;

export const ButtonIcon = themed(styled(Icon)`
  display: flex;
  font-size: ${({size}) => `${size}px` || "48px"};
  margin-bottom: ${({bias}) => bias || "0"};
  color: ${({type, theme: {buttons}}) => buttons[type].text};
`);

export const MenuButtonText = styled.Text`
  font-size: 12px;
  color: ${({type, theme: {buttons}}) => buttons[type].text};
`;

export const TabContainer = styled(ButtonContainer)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  min-width: 142px;
  width: 100%;
  max-height: 38px;
  margin: 0;
  padding: 10px 30px;
  border-bottom: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
`;
export const TabText = styled(BtnText)`
  top: 0;
  color: inherit;
`;
