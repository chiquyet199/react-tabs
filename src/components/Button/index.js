import React from "react";
import {View} from "react-native";
import {
  array,
  bool,
  func,
  object,
  oneOf,
  oneOfType,
  number,
  string,
} from "prop-types";
import {
  ButtonContainer,
  Left,
  Middle,
  Right,
  BtnText,
  StyledIcon,
  IconContainer,
  MenuButtonIconContainer,
  BounceContainer,
  MenuButtonText,
  ButtonIcon,
  TabContainer,
  TabText,
} from "./Button";
import Icon from "../Icon";
import {applyStyleObjectToStyledComponent, themed} from "../../utils/common";

export const Button = ({
  left,
  right,
  middle,
  onClick,
  type,
  noborder,
  style,
  iconsize,
  theme,
  disabled,
}) => {
  const color = theme && theme.buttons[type].text;

  return (
    <ButtonContainer
      noborder={noborder}
      type={type}
      onPress={onClick}
      activeOpacity={0.6}
      style={style}
      theme={theme}
      disabled={disabled}
    >
      {left && (
        <Left size={iconsize || 32} shift={middle}>
          <StyledIcon
            type={type}
            name={left}
            size={iconsize || 32}
            color={color}
          />
        </Left>
      )}
      {middle && (
        <Middle lefticon={left} righticon={right} size={iconsize || 32}>
          <BtnText ellipsizeMode="tail" type={type} theme={theme}>
            {middle}
          </BtnText>
        </Middle>
      )}
      {right && (
        <Right size={iconsize || 32} shift={middle}>
          <StyledIcon
            type={type}
            name={right}
            size={iconsize || 32}
            color={color}
          />
        </Right>
      )}
    </ButtonContainer>
  );
};

Button.propTypes = {
  left: oneOfType([string, bool]),
  middle: string,
  right: oneOfType([string, bool]),
  onClick: func,
  type: string,
};

Button.defaultProps = {
  left: null,
  right: null,
  type: "primary",
};

export const IconButton = ({
  onClick,
  name,
  type,
  style,
  iconsize,
  bias,
  theme,
}) => (
  <IconContainer style={style} type={type} onPress={onClick}>
    <Icon
      size={iconsize}
      type={type}
      name={name}
      bias={bias}
      color={theme && theme.buttons[type].text}
    />
  </IconContainer>
);

IconButton.propTypes = {
  name: string.isRequired,
  onClick: func,
  type: string.isRequired,
};

const ThemedIconContainer = ({
  name,
  type,
  style,
  iconsize,
  bias,
  theme: {buttons},
}) => (
  <IconContainer size={iconsize} style={style} type={type}>
    <ButtonIcon
      size={iconsize}
      type={type}
      name={name}
      bias={bias}
      color={buttons[type].text}
    />
  </IconContainer>
);

ThemedIconContainer.propTypes = {
  name: string.isRequired,
  type: string.isRequired,
  style: oneOf([array, object]),
  iconsize: number,
  bias: number,
  theme: object.isRequired,
};

export const StyledIconContainer = themed(ThemedIconContainer);

export const MenuButton = ({
  onClick,
  name,
  type,
  iconContainerStyle,
  children,
  iconsize,
  theme,
}) => {
  const StyledInnerContainer = applyStyleObjectToStyledComponent(
    iconContainerStyle,
    MenuButtonIconContainer,
  );

  return (
    <View style={{flex: 1, alignItems: "center"}}>
      <StyledInnerContainer type={type} noBorder onPress={onClick}>
        <Icon
          type={type}
          name={name}
          size={iconsize || 32}
          color={theme.buttons[type].text}
        />
      </StyledInnerContainer>
      <MenuButtonText type={type}>{children}</MenuButtonText>
    </View>
  );
};

MenuButton.propTypes = {
  onClick: func,
  name: string,
  type: string,
  iconContainerStyle: object,
  iconsize: number,
  theme: object,
};

export const FormBounceButton = ({children, onClick, globaltheme, theme}) => (
  <BounceContainer>
    <Button
      onClick={onClick}
      middle={children}
      type={theme[globaltheme].form.bounceButton}
      noBorder
    />
  </BounceContainer>
);

FormBounceButton.propTypes = {
  children: string.isRequired,
  globaltheme: oneOf(["dark", "light"]),
  onClick: func.isRequired,
};

export const TabButton = ({onClick, left, iconsize, middle, type}) => (
  <TabContainer type={type} onPress={onClick}>
    <StyledIcon type={type} name={left} size={iconsize || 20} color="inherit" />
    <TabText ellipsizeMode="tail" type={type}>
      {middle}
    </TabText>
  </TabContainer>
);

TabButton.propTypes = {
  left: string,
  type: string.isRequired,
  middle: string.isRequired,
  iconsize: number,
  onClick: func,
};

export default themed(Button);
