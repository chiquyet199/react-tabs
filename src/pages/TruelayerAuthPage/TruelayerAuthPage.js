import React, {Component} from "react";
import {withTheme} from "styled-components";
import {
  TruelayerContainer,
  TruelayerPageTitle,
} from "./TruelayerAuthPage.styles";
import {webhooks} from "../../constants/webhook";
import {StyleSheet, WebView} from "react-native";

export class TruelayerAuth extends Component {
  render() {
    const {translations, theme, globaltheme} = this.props;

    return (
      <TruelayerContainer>
        <TruelayerPageTitle globaltheme={globaltheme} theme={theme}>
          {translations.truelayer_auth_title}
        </TruelayerPageTitle>
        <WebView
          style={styles.webview}
          mixedContentMode="always"
          title={translations.truelayer_auth_title}
          onMessage={ev => ev.nativeEvent}
          // FIXME Add to .env
          source={{uri: webhooks.truelayer.callbackUrl}}
        />
      </TruelayerContainer>
    );
  }
}
// Should be locked inside Native Component (because styled.WebView fails on Web during tests)
export const styles = StyleSheet.create({
  webview: {
    flex: 1,
    justifyContent: "center",
    alignItems: "stretch",
    alignSelf: "stretch",
  },
});

export default withTheme(TruelayerAuth);
