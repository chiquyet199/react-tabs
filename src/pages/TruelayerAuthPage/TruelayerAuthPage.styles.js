import styled from "styled-components/native";

export const TruelayerContainer = styled.View`
  flex: 1;
  justify-content: space-around;
  align-self: stretch;
  align-items: stretch;
  flex-direction: column;
  flex-wrap: wrap;
  width: 100%;
  max-width: 100%;
`;

export const TruelayerPageTitle = styled.Text`
  font-size: 28px;
  line-height: 36px;
  margin-bottom: 20px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  width: 100%;
  text-align: ${props => (props.left ? "left" : "center")};
  padding-left: ${props => (props.left ? "15px" : "0px")};
  align-self: ${props => (props.left ? "flex-start" : "center")};
`;
