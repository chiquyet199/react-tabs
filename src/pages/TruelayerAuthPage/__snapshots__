/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/pages/TruelayerAuthPage/__native_snapshots__