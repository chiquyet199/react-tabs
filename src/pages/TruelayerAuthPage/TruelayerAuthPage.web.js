import React, {Component} from "react";
import {
  PageContainer,
  PageTitle,
  ButtonBar,
} from "../../components/common/layout";
import styled, {withTheme} from "styled-components/native";
import Button from "../../components/Button";

export const StyledButtonBar = styled(ButtonBar)`
  margin-top: 40px;
`;

export class TruelayerAuth extends Component {
  saveUserToken = async () => {
    // FIXME add functionalitty later depending on implementation
  };

  handleIncomeMessage = () => {
    // FIXME add functionalitty later depending on implementation
  };

  render() {
    const {translations, openAuthWindow, globaltheme, theme} = this.props;

    return (
      <PageContainer>
        <PageTitle globaltheme={globaltheme} theme={theme}>
          {translations.truelayer_auth_title}
        </PageTitle>
        <StyledButtonBar>
          <Button
            type="primary"
            middle={translations.authorize}
            onClick={openAuthWindow}
          />
        </StyledButtonBar>
      </PageContainer>
    );
  }
}

export default withTheme(TruelayerAuth);
