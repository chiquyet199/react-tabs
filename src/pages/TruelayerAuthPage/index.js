import {connect} from "react-redux";
import TruelayerAuthPage from "./TruelayerAuthPage";
import {webhooks} from "../../constants/webhook";

export const _openAuthWindow = () => {
  let w;
  let h;
  if (window.outerWidth > 768) {
    w = window.outerWidth / 1.8;
    h = window.outerHeight * 0.9;
  } else {
    w = window.outerWidth * 0.8;
    h = window.outerHeight * 0.8;
  }
  const y = window.outerHeight / 2 + window.screenY - h / 2;
  const x = window.outerWidth / 2 + window.screenX - w / 2;
  const params = `scrollbars=yes,resizable=yes,status=no,location=no,toolbar=no,menubar=no,width=${w},height=${h},top=${y},left=${x}`;
  // eslint-disable-next-line no-restricted-globals
  window.open(
    webhooks.truelayer.callbackUrl,
    "Please choose your bank",
    params,
  );
  window.focus();
};

export const mapStateToProps = state => ({
  openAuthWindow: _openAuthWindow,
  translations: {
    truelayer_auth_title:
      state.locale.translations.truelayer.truelayer_auth_title,
    authorize: state.locale.translations.authorize,
  },
  globaltheme: state.globaltheme,
});

export default connect(mapStateToProps)(TruelayerAuthPage);
