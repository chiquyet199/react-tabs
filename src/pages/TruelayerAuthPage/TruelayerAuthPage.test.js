import React from "react";
import {shallow} from "enzyme";
import {initialState} from "../../__mocks__/initialState";
import {TruelayerAuth} from "./TruelayerAuthPage";
import configureStore from "redux-mock-store";
import Button from "../../components/Button";
import locale from "../../utils/locales/en-US";
import {isNative} from "../../utils/common";
import {WebView} from "react-native";
import {_openAuthWindow} from "./";

const mockFunc = jest.fn();
const mockProps = {
  translations: {
    truelayer_auth_title: locale.truelayer.truelayer_auth_title,
    authorize: locale.authorize,
  },
  openAuthWindow: mockFunc,
};

const mockPropsWithWWindow = {
  translations: {
    truelayer_auth_title: locale.truelayer.truelayer_auth_title,
    authorize: locale.authorize,
  },
  openAuthWindow: _openAuthWindow,
};

const mockStore = configureStore(mockProps);
describe("Testing TruelayerAuth snapshot", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TruelayerAuth {...mockProps} />, {
      context: {store: mockStore(initialState)},
    });
  });
  it("renders as expected", () => {
    expect(wrapper).toMatchSnapshot();
  });

  if (isNative) {
    it("renders webview", () => {
      const webview = wrapper.find(WebView);
      expect(webview).toHaveLength(1);
    });
  }

  if (!isNative) {
    describe("functionality", () => {
      it("trigger submit callback", () => {
        const btn = wrapper.dive().find(Button);
        expect(btn).toHaveLength(1);
        expect(btn.prop("middle")).toBe(locale.authorize);
        expect(btn.prop("onClick")).toBe(mockProps.openAuthWindow);
        btn.simulate("click");
        expect(mockProps.openAuthWindow).toHaveBeenCalled();
      });
    });

    describe("check _openAuthWindow method", () => {
      const tree = shallow(<TruelayerAuth {...mockPropsWithWWindow} />, {
        context: {store: mockStore(initialState)},
      });
      it("triggers open and focus", () => {
        global.open = jest.fn();
        global.focus = jest.fn();
        const btn = tree.find(Button);
        btn.simulate("click");
        expect(global.open).toHaveBeenCalled();
        expect(global.focus).toHaveBeenCalled();
      });
    });
  }
});
