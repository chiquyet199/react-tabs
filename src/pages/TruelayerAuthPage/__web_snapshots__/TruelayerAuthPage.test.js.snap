// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TruelayerAuth snapshot renders as expected 1`] = `
<Connect(ResponsiveWrapper)>
  <Connect(WithTheme(Styled(Text)))>
    Truelayer Authorize
  </Connect(WithTheme(Styled(Text)))>
  <Styled(Styled(View))>
    <Connect(WithTheme(Button))
      middle="Authorize"
      onClick={[MockFunction]}
      type="primary"
    />
  </Styled(Styled(View))>
</Connect(ResponsiveWrapper)>
`;
